ui/react-components/components/AppointmentEditor/AppointmentEditor.jsx

```diff
@@ -179,44 +283,146 @@ const AppointmentEditor = props => {
+                                onChange={date => {
+                                    setRecurringStartDate(date);
+                                    setStartDateError(!date);
+                                }}
+                                onClear={() => setRecurringStartDate(undefined)}
+                                startDateType={startDateType}
+                                endDateType={endDateType}
+                                dateType="startDate"
+                                startDate={recurringStartDate}
+                                isRecurring={isRecurring}
+                                defaultValue={recurringStartDate}/>
+                            <ErrorMessage message={startDateError ? dateErrorMessage : undefined}/>
                         </div>
-                        <div data-testid="end-time-selector">
-                            <TimeSelector {...appointmentEndTimeProps}
-                                          onChange={time => {
-                                              setEndTime(time);
-                                              setStartTimeBeforeEndTimeError(!isStartTimeBeforeEndTime(startTime, time));
-                                          }}/>
-                            <ErrorMessage message={endTimeError ? timeErrorMessage : undefined}/>
+                        <div data-testid="end-date-group">
+                            <div className={classNames(dateHeading)}>
+                                <Label translationKey="ENDS_LABEL" defaultValue="Ends"/>
+                            </div>
+                            <EndDateRadioGroup
+                                onChange={event => {
+                                    setEndDateType(event.currentTarget.value)
+                                    event.currentTarget.value === "After" && setRecurringEndDate(undefined);
+                                }}
+                                onOccurencesChange={value => setOccurences(value)}
+                                occurences={occurences}
+                                endDateType={endDateType}/>
+                            <AppointmentDatePicker
+                                onChange={date => {
+                                    setRecurringEndDate(date);
+                                    setEndDateError(!date);
+                                }}
+                                onClear={() => setRecurringEndDate(undefined)}
+                                isRecurring={isRecurring}
+                                startDate={recurringStartDate}
+                                startDateType={startDateType}
+                                endDateType={endDateType}
+                                dateType="endDate"
+                                defaultValue={recurringEndDate}/>
+
+                            <ErrorMessage message={(endDateTypeError && endDateTypeErrorMessage) || (endDateError && dateErrorMessage)
```

Comment:
extract into method to avoid warnings in console.

Reply:
yes. Agreed